src/Deck.java

```diff
@@ -0,0 +1,69 @@
+import java.util.Random;
+
+
+public class Deck {
+
+    private Card[] deck;
+    private int currentCard; //index of next card to be dealt
+
+    public Deck() {
+
+        String[] faces = {"2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"};
+        String[] suits = {"Spades", "Clubs", "Diamonds", "Hearts"};
+
+        deck = new Card[52];
+        currentCard = 0;
+
+        /**
+         * Building the deck
+         */
+        int suitCounter = 0;
+        while (suitCounter < 4) {
+            int faceNum = 0;
+            while (faceNum < 13) {
+                deck[(faceNum + (suitCounter * 13))] = new Card (
+                        suits[suitCounter],
+                        faces[faceNum],
+                        faceNum+2);
+
+                faceNum++;
+            }
+            suitCounter++;
+        }
+    }
+
+    /**
+     * Printing the deck
+     */
+    public void displayDeck() {
+
+        for (Card card : deck)
+            System.out.println(card);
+
+    }
+
+    public void shuffle() {
+
+        Random mrRandom = new Random();
+        int currentCard = 0;
+        int first = 0;
+        while (currentCard < deck.length) {
+            int second = mrRandom.nextInt(52);
```

Comment:
what can ensure that no duplicates here? there is chance that `mrRandom.nextInt(52)` would return duplicates.

Reply:
yes, there might be duplicates but in the way that I intended to use this, I just wanted to switch the cards around with each other. So even if the random returned a duplicate, the cards would just switch around. For example, I'm switching the card in index 0 with index 32. Then the next index 1 could switch with index 32 again but the cards are not repeating and will still end up shuffled.

Reply:
good point